ARTEMIS-2141 Add Filter column to Consumers View of AdminUI

sebthom opened the pull request.

michaelandrepearce commented on artemis-server/src/main/java/org/apache/activemq/artemis/core/management/impl/view/ConsumerView.java:
```diff
@@ -89,6 +90,8 @@ public Object getField(ServerConsumer consumer, String fieldName) {
          case "queueType":
             return consumer.getQueueType();
+         case "filter":
+            return consumer.getFilter() == null ? "" : consumer.getFilter().getFilterString();
```
Rather than the dupe logic here and on line 61, create a method to getFilterString.

Also if null, should return null, not empty string.
         return filter != null ? filter.getFilterString().toString() : null;

e.g. should be same logic as in QueueControlImpl

michaelandrepearce commented:
Please also update ConsumerFilterPredicate to handle the new field, see QueueFilterPredicate for example of handling filter in the filterpredicates

sebthom commented:
@michaelandrepearce I updated the ConsumerFilterPredicate too now and also changed the filter related implementation of the getField() method in ConsumerView according to QueueView.

michaelandrepearce commented:
@sebthom LGTM, thanks for the enhancements and the contribution

sebthom commented:
@michaelandrepearce Thanks for merging the PR! Is there any chance this gets backported/included into 2.6.4?
